lib/internal/Magento/Framework/Css/PreProcessor/Instruction/MagentoImport.php

```diff
@@ -106,24 +135,36 @@ protected function replace(array $matchedContent, LocalInterface $asset)
             $relatedAsset = $this->assetRepo->createRelated($matchedFileId, $asset);
             $resolvedPath = $relatedAsset->getFilePath();
             $importFiles = $this->fileSource->getFiles($this->getTheme($relatedAsset), $resolvedPath);
+            $deployOnlyEnabled = (bool)$this->deploymentConfig->get(self::CONFIG_PATH_SCD_ONLY_ENABLED_MODULES);
             /** @var $importFile \Magento\Framework\View\File */
             foreach ($importFiles as $importFile) {
+                $moduleName = $importFile->getModule();
+                
+                if ($moduleName && !$deployOnlyEnabled && !$this->moduleManager->isEnabled($moduleName)) {
+                    continue;
+                }
+                
+                if ($moduleName && $deployOnlyEnabled && !$this->moduleManager->isEnabled($moduleName)) {
+                    continue;
+                }
+
```

Comment:
added 2 case check here
@ihor-sviziev . What do you think? Feel free correct me

Reply:
If the flag isn't available - it should work the same as on 2.4-develop. Otherwise we'll have backward incompatible change

Reply:
updated code can you review one again